### PR TITLE
HDDS-9065. [FSO]ListKeys: Incorrect result when keyPrefix matching multiple exist keys.

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Fix  cases when the keyPrefix pertains to a particular file/dir  . For more details on the problem check jira description.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9065

## How was this patch tested?
Unit tests.

### CODE REVIEW

**Comment by sadanand48** [comment]
@whbing would you like to take a look?

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java [hunk 1]
```diff
@@ -81,6 +81,7 @@ OzoneFileStatus apply(OmKeyArgs args, String clientAddress,
     this.omDefaultReplication = omDefaultReplication;
   }
 
+  @SuppressWarnings("methodlength")
```
[the remark this answers is not in the document]
Comment (TaiJuWu) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java:84]: Is there a good tool which can help us format this project?

[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java:84]: Maybe even can be split into several methods (validate parameters, handle different cases, collect results, etc.).

[the remark this answers is not in the document]
Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java]: Thanks , I have refactored the method now.

**Comment by sadanand48** [comment]
> I'm not sure if there is a similar problem in listStatusLight

Thanks for the review @whbing , listStatusLight calls listStatus internally, It only trims the keyInfo object to be sent over the wire.

**Review by aryangupta1998 — APPROVED** [review]
LGTM

**Comment by aryangupta1998** [comment]
Thanks @sadanand48 for the patch, @TaiJuWu, @adoroszlai, and @whbing for the reviews!

**Pull request merged**